### PR TITLE
Geostyler install script

### PR DESCRIPTION
**Pull request by jansule**
This installs the GeoStyler libraries as well as a beginner tutorial to the OSGeoLive.

### CODE REVIEW

**Comment by jansule** [comment]
Open for review

File: bin/install_geostyler.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+#
+# Install desktop icon and demo
+#
+
+mkdir -p /usr/local/share/icons/ && cp -f "logo.svg" \
+       /usr/local/share/icons/geostyler.svg
+
+#
+# Launch script and icon for GeoExt
```
Comment (marcjansen) [inline bin/install_geostyler.sh:125]: Comment copy and paste error

Comment (marcjansen) [inline bin/install_geostyler.sh:125]:
```suggestion
# Launch script and icon for GeoStyler
```

**Review by marcjansen — APPROVED** [review]

**Comment by cvvergara** [comment]
Related documentation https://github.com/OSGeo/OSGeoLive-doc/pull/604

**Comment by kalxas** [comment]
First nightly build logs: 
http://aiolos.survey.ntua.gr/gisvm/dev/osgeolive-nightly-build112-amd64-53c349b-master-log.tar.gz

Unfortunately enabling GeoStyler broke the build, so I am going to disable it for now

**Comment by kalxas** [comment]
Disk usage: 430 MB

disk_usage_plot

**Comment by jansule** [comment]
@kalxas According to the logs, the installation of GeoStyler seems to be successful. But when creating the iso, the overall size (of all applications) seems to be too big (https://trac.osgeo.org/osgeolive/ticket/467#comment:51).

So GeoStyler "just" seems to be the unlucky application with which the size limit was reached.

Do you know how much free space is available, when not installing GeoStyler? I could try to improve the disk usage, but a rough estimate would be nice to have.

**Comment by kalxas** [comment]
Unfortunately the disk size available is zero, we are trying to slim down the iso.
I have opened a ticket to follow this:

https://trac.osgeo.org/osgeolive/ticket/2269

**Comment by kalxas** [comment]
@jansule the problem is that GeoStyler seems to take over 400MB of disk space.
Is there any way to make it smaller?
Another issue I see during the build is the long npm execution time, I am waiting over 20 mins...
Can we just add a pre-compiled, slimmed down version in the installer?

**Comment by kalxas** [comment]
Opened ticket:
https://trac.osgeo.org/osgeolive/ticket/2269

**Comment by jansule** [comment]
@kalxas There are a few ways we could handle this:

*The disk usage saved is just a very rough estimate, but should give you a brief idea about how much usage is reduced*

**remove the geostyler lib and nodejs, and just keep the tutorial (~390MB saved)**

This would reduce the disk usage drastically. However, users would have to install nodejs and the geostyler themselves. Especially with regard to the workshops we offer on the German FOSSGIS, this usually takes a lot of time, and poses a difficulty for unexperienced users.

**use pre-compiled version of GeoStyler, keep tutorial (~350MB saved)**

This is technically possible. However, this is not the common approach to work with react.js libraries/node applications, and I think this should not be taught to potentially very unexperienced users, as the workflow differs strongly from the recommended approach.

**remove the geostyler lib, keep nodejs and the tutorial (~200MB saved)**

IMHO the best solution. Installing geostyler manually is not a big problem, if nodejs is already installed on the system. Having nodejs on the system would also be a big advantage for all javascript based applications/libraries.

> Another issue I see during the build is the long npm execution time, I am waiting over 20 mins...

That surprises me. When I wrote the installation script, it did not take that long

**Comment by kalxas** [comment]
Thanks for the feedback @jansule 
I would go with the second option if there is a demo application for the users to test. 
If the users are supposed to only do development, I would go with option one.

**Comment by marcjansen** [comment]
Option 3 is best IMO. NodeJs (wherever it comes from) should be on osgeo live.

**Comment by kalxas** [comment]
The first goal of OSGeoLive is to demonstrate the included projects. Does GeoStyler have a demo application to test on startup?
I will test how much disk space is used by including nodejs by default.

**Comment by jansule** [comment]
Yes, we do have a demo application (https://geostyler.github.io/geostyler-demo/). For a library, however, I think it's more important that users can have a quick hands-on on the code.

> I will test how much disk space is used by including nodejs by default.

It might be helpful to take a look at [nvm](https://github.com/nvm-sh/nvm), as this can be used to install any version of node. We also used it in our installation script.

**Comment by kalxas** [comment]
We have a strong preference in using the default ubuntu packages. Does geostyler work with nodejs 10?

**Comment by kalxas** [comment]
We also had a project meeting last night and we would like to include the demo application of GeoStyler and include the development part in the quickstart.

**Comment by jansule** [comment]
> Does geostyler work with nodejs 10?

Yes, it does.

> We also had a project meeting last night and we would like to include the demo application of GeoStyler and include the development part in the quickstart.

Thanks for the info. I will update the installation script + docs accordingly soon.

**Pull request merged**